### PR TITLE
db: LSM viewer URL generation

### PR DESCRIPTION
**Pull request by RaduBerinde**
This commit adds code to generate a URL that displays a diagram of the
LSM. This is primarily meant for debugging, in particular metamorphic
test failures.

[Example](https://raduberinde.github.io/lsmview/decode.html#eJzUnU-PJMdx9u_vp2jMe10J8cSfJyLm7ptv9s0ihBU0EAStKJtLGpYFfncja3p2u6vau8PsXGMIAkMSDfyQlZUZT0RGZNQ_Hj48_efTh48Pj_92_s_ff__-r08Pjw__LA_vHn58_4cPT-PH735-t_8dX_ldv_K7feV3_8rvcfn7Px4-vP_D04eHxwfmw7uHj3_-76eHR1OTiH738PGv7z98ePr44-__8vT3h0e8e_jw_oc_ff5_effwx6cf3_95m4YHERHm4-mnj08_QEQ8OrtJJwnt-t3v_kvklX8w_vT_Z5XJu3_5p389_eYZG5alAbSGtEQxbQbbYTGw44mf_uP7n_768fFE1dbTb06J6vEK__1vf_7-x4-_tsc5fZPxXoMJgq0NLWRgEmwHcNKt0Z3m1HKfAmf5AdzVMJFI0RJ0zIAt5QCGhHQJzDpDDDoDpshhKqAZ2z-eZl3CGbCSeQCbZ1hJqidN-VrmFdjF-gBmdkt6J9uN3lPrGH2YClW2pqRoOFQxM-KE1w1wQiqbKWJKTu28sd4OYOvqqGEMTZNTq62iDxtPvUSqI0UYkzMcYjxwk8qE0zs81XJqTcRxFWuZe6Cp7BT2lKWw5Gfwb3_72-3f3__t9MP77__0dPrjkOBnjXuRsbqQsQy1PMiY7XRM7ahj9fjZQreYh2rBVKdWSWbhwuw7gj5WnApRpqkzayRh0rdEzJ5FrHovYr-OZzl9g8FeYgvaCURZFEUxtY_o4AHsQvcSItKKNmW0rfQGGNTuVFF4sqY2aCoOM1xsk4zI1uxSnwKr-XGOE0MMsspNK2vGaJNSuge3tmpAhgMilpia4yqXA9gzOqTDxQibky87TnFHs9ys3VJCOecipB8WRZciCh4uoUpM7Q9L-BFc5oYw6W5Nm3x3t8BtRnQg2ymc0xkc9ocLsrva2SESzClX16T2L88loBZV7YmaHHEKfqGA9YWASTqxFzCNnYCZHwWsH1_sc3u4laa7wAozs5OC1Aur32HammGkR23mdHIj3gzDcA7DsFOwX8vDnL7FaC-4VFNROiAm0Jqygh278MBBw9igKI6NSp_bSHYcL4f3yhQXV2hMWVfv6D03s7VULbWqO6femyf2401IZ4eAzczk1HsDdD-_Cc9wNDDifplziMuSe64C3Rapri1zQV2Eyh7b2tkV3hUamIrHNW0_uyUaBgmkISrmjjyGrB64laBEu1KLPTUNxjhwtbJhjhR0x9zpAWw_u2VAdXH4Hk7xqcXATBy4WlIRFWHW5JRxoF0Yh1eIVson0dJq04NmjTDsWrP6oFkpjy8GOTpCK8ejjJ0yd3iVz2dML2a-uxPDgaQgzXPKClkDR8kycJOsbNtJ1q_kWU7fYrDX3DSPEcSwi5yzfLk75uwwRiM4XO-x1m1qc4bu9nyHlRVaVWljh84dcoYc5tdRTOkRyYGpOhUOZMd-ft3TtNOs0sMnbR_RPHIDcMpwRQM-Y6tTtPfvzVOzrGFbJDAXxjVljw0RWieLVpY-tRq0jtgxAylRZhSXOZPqLvvZDZYFHKYSETUXf4ek7Lmd3W0m5aIeMhfNOvZcBtuCAVNvk5xZZQnUfrcxEkjLzAQsJs965GKV_W-SdSMdxtvpMLdPiuboLTrcSZrsjxEPgub2OX0Eba0RJPQwYzKzNK9SR5AcDkQarQOimLKpstcyOf3mdCv79baHf1o-zmuop0INtZ1n9cw2vQ0NsLuJcJ_ZSzegY3NmhmZhrNll0EyYVVSozWjWDWiZ0FBkllYtevtlEA1mRoToTMRyhKpAtLZpUK8prboNZVpparOnIuIbUFS3Frjl_qaclhtQY1l7lbFVY83i1_ByY7mr0KbcthuJabOyruETivaU73obmq3IMaNy55y-IpJyv9QdFz3kr_bpq-Phn_vjZyurndFsNe2uqfTpleFWhBm121JE1VzvWbtf1p03P_zT8nFeQz1z-G9p4phzZ4_QUCZFXCIgvkJ2B5QaidBUZrYskIhnKCEtZR4iqx6_xJtNgJ7sBRLxDA2juaqjayqDfhuakWqGRk_l0G5Au1zVOoRInzo5vg0lMgmr5qIdFV0jgI4cvmHmmiVF0CoVIhbtumbx09ygdGvT1qlM3G1oqGgIBBb3vajX6E7sdMf2urPPOvGoO_H4ycqGDUuYKHVzmV-9L4bbVJuM4TVIqJXnPbvsy7rz5od_Wj7OaygbgA5DKYgVu-QMDYmiWBq5wPAOaKUYaVWRsBVb7wxNCrrNze_cehdQbLnqYdOiF9hIRVBCfOhjS4VN5ZNuQFVbWZWpZbIiMB1Ql2qkaIr6XKbuBpSILniMCZg6Sz0yC92aOuShba5m7QbUpUbw6IaqqUTaLSbUqR3S0ivCssGMyKGPShK4M9J_jehwJzqxF5192qiOosPHSxNLtXRwhNc6vXRfrLabgCmRYBbpWfdMyZdF580P_7R8nDto97BnIVpzqfEbzBQLN0SJqC3Yd4NZHt7I8C7xqZrvA9REdLh7WoYsWRGUnaHplB7uvsqSxzdRMQqpCs-KBYr7DPXtrEUb3b1AcbcrLFQgTEQqfUVQZmrwVgxD3p1Va54ew88qhmeK6JqHR8hWcxPhCi7wNkxNLdRTaBpBLHr1GoYcLgFCo-8M8l8jOXmd1zlKTu-vOeGoOS_XnM4Wthle7caeKn8-3AmyMTEuqdaIu7T9y5rz5od_Wj7OK6hpQbOKGCOdugJyAzpMrqYAMPOpi1c3oIV2C1ZLmqzQ8QHtLrYMwXWtXCAQz9CWsrIqhGDN23eMVVXIzPKaSmjfgFokulKiSz3WvKgBjdYAM5yyQMkHNIzlHO65uvWadephqS4kvRlT1xVuQbMTAncfYckaZoK-1XY7abrmPZEVCSkSprxzSl-jO3WlO5CD7kD3wnMjs3N5LymAyHLACJu6D3ZlubfqZwsvq-HeecS3C3be_PBPy8d5Da0cXqlUm0jbgg0dVkKOFUJId8bUpdHbUHd1RbLnbh0eoRBstqyqLWWqWu02tGN7eGldNKdAZwHUbGGt0N1naKmEWzFbVvgyZ2h0q7cHMFWzfgNqmR6dmfSqFWJ-hlZYVLlKT5Wr34BGMOEAieIKD2FA07b7WxFjsCvE_BlKoypplJi65v7LlKevlKd5bOywT-3gRm7nfKFo2NlmdiO9qzRleqO9WO4Mk05vi4LRMVX_9zrdeeODP60e5Y5ZmW0JV_aK2jA6JTXc47lGBitM2Rla1hjRiekKU0YnpKzDhVGUXGEgnqFunpWbY7rC4xhQpamUi5dJrLA6ZyhFt6SR8k6rcwntHkDLMQtroO6mIcYwH-HEImjSurqpTlu0oqgKyOZ1qq6ISwa0BBktblqOFUcSdI5N1K5W0e5TN3p_keKEXJ-x5THW2ed1cEzsxPk60DCxYDg9XTO05m74XlptFSmV4YTSmep6V5L3y5Lz1kd_Wj7MK6gKNKnavhVdLdjNA4pWh7i7W0ksMGZnaIlrhEKwosp7QC2xZd7MJcvXvCi1VImxmZ9vvq-BRkijoWqZS0zkMxQGbTP4CKTWQNO2W28jHM97zdklNJi63aWdu5R6CzpCkjRAKCsOGDPMgyKq0SaWsmaVemXLkHBkxFzLhyN0K_apoI5Xz6k-SL9MdXClOgk9qM4-taPHDnaBx09G1rLFObxwRd9958UkWSKVEUoU4q4D9y-Kzlsf_Gn1KPdMt-4K9vChlzAjysbisGpwgTYMZlpGeA2xhS-w4mdme5FGMy64qqMiPfzQrOhI9IJ01jMyCKaYO6Zuvt9C9phNANRasTghodnbUaWZ6JLFCamio2q45J1LXvoQFwpTYALkgltPZ6ZTSi3LaqrzzZGpPrwsC3QhFlykOjNzwLYGLnfeo3qNyOi1yNAP3Xn2bVL1eDE09PGTVe0SaBcDAZu6Wnxppx1B0wLCM1yN305k3vrgT6tHecnM8GSqqDlWVIluyAAhMMBqQbGaSVameAU6uQWNK5gNrW4bnkBMthc5MsdYw10sI22q_9ItZqmXwkN7RXZ_Y7oowsc78gXVXwPptfUhdU1NLPDnB5ObAyAFNmtBrfmZmcFqdbb4mqVUTpcQV4X0ghsR45FhmlElW2nPipU0kGSzIWE-1cTrl0mM7STmcANUfS8xx3xN2KeeaQxJYxfc2vxuKx1qLgxra9Fm610n1F-UmLc--NPqUe6ZSaMGGCsymoPJopqYazTmGt3dYqYgmw6LFUnSwSyGl7MwnnyBVTwzK8tbzUzitbivMak5ZhQSWFCY5YgUH0E2XSW9a8k7arR1sxwJYIH1fma6tWxdEUwXjdPSNIASlvQCJ-jMpHs3PIabtYQZoSER6LFLF9Rsn5m1te1x1bqzZvs1OnPVacBKDnUBus_S6I0szbnXQKhtLeciXJ2MnPfdz6aaJjki8HJPNO2uLPgXZeaNj_20epCXTIqFiEW6S65Iiw-mwooOUUrYAtd7MK0NKB8GMrDAep-ZAdXhFKQusN6DGalJRsOQssAqPjMNFBeV6hWJ5sHMUIenQ6JlwR2uwWwD2VIiWrXgalSoJSA9ouzqLi64xBS6dQCgCQWpYWve0XhqWOsIlLwX3LgZTI22NqJongtuhZ2ZOcLNdhmB3F3M16hMXEczQ0J2KrPPytiNrMy5swCf8xClPfwPVZ2vOTlb6rQCkAKRpEnEXUnKL8rMWx_8afUod8wBa3bdX4PywhwGIdO2Q_W5ZpoHZon1sLbPKfsFd7UHMzKryjNi69iwgAlBlQrK00f0teK9b8zoYJS62pJnhxhHzNWdCsWCovTBDE-I27CMqAUpqTPTRwDPDp_rxnxkpm4yW9Fqtei9t1CEpV5tXFBwSQMAk4QGLGrJdgfcHTSGDnft2xc5B6-DmRtt0_Z5GbuRlzn3Ehh2lRGbM2NdiVeP_fAAZ0tdHh2pI6YtaHpMfQnkVTLz1gd_Wj3KPTMoW6f91ql-xUdmmbtCXXs7AF_FLISEpbYsaICYNuRguPVRaCUX5MAHc-sy3j0iD3reb76fmWZqQ2VZXFASOJhjHnPrrgKJBRWBgzl0IOBRHljQsOgZWVIdJNIWHGvmc8dLrdAwY2DN6nQ02V5e8F5QXTqQKkLT3rorLUhJDWQo3FUK4kzcJ9mv0ZjchTLHrxzse3NaHDXm3DugnEBpiZC-fZnx3rsoHWao3Pr3p2vOfcDlVRrz1gd_Wj3KK6ZR3atg6aYL6qQHMxw2BBbt0nq_Zdg6MqoEjSUNyQV29szsSG-wtBb4AYPZIoqtIXQoF9juM5MSdLh6LijYK6dKtuT2RbCQFUe6Z2ZF9wg7PRbUgw2mRpmSnaWy4hL-mTmWk4xYRhdUhw-mW9mWlTNzLCi5LqerVqp1S4nWmvn0KNbWYrBT7P_g-sznVgEoWPihkNm415kbiZl6vDSsrRpe6M75aqWLr8-Yjtiuq3X42vdMyRd15q0P_rR6lFdMYyJZlWDHgnKjDotA0FOKJZFrnn1EceJSnds515pxsr0FrpEZsaCQ-ZkZ2ls-xZMLaoPOTFaLp6HnvvB3ZFY6IpKeHragnvfMbDcrajUX9CfvMErY0C24V8591e8W09GZunUDW9BIfDDVTKI0zCPvbeF_wfRQGYLTWNBGvMMquFV_wCg591XHr-vMdz9_9-7hL09_33Tip4-rvpry8O4TzLfPD5FOEjpzOvsCW_KRmEvY3U36X2BLPiNwAbu_jfMLbEmj6WvYnb0-X2BLupHuYPd1hHuBhWVpAK0hLVGc8UUvYXc3wLuE3d1_6BJ25yfVX1BLmi1dwO5ud_HCWtGO44J1_5XoM2zJne0L1t335s6sJff6Llh3X7U4s5ZcBblguZKmQpRp3mErltT6fmbd_4HoM2tJXfNn1v2VZmfWikK4C9TdhQln1pLCiQvW3ZmsM2tJpu0Ta8Hp55m15Bvpn1n3n_Res-6LvD-z7v_67QtrxSnDBevej_0-fPfz__ufAAAA__83sgVg)

### CODE REVIEW

File: internal/lsmview/url.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+	if err := compressor.Close(); err != nil {
+		return url.URL{}, err
+	}
+	if err := encoder.Close(); err != nil {
+		return url.URL{}, err
+	}
+	return url.URL{
+		Scheme:   "https",
+		Host:     "raduberinde.github.io",
```
Comment (itsbilal) [inline internal/lsmview/url.go:37]: This works for me, but do we wanna move this base64 decoder to `cockroachdb.github.io` at some point?

**Review by itsbilal — APPROVED** [review]
LGTM

File: internal/lsmview/url.go (hunk 1 above)
Comment (RaduBerinde) [reply internal/lsmview/url.go:37]: Yeah, if this turns out to be something we like and want to keep, we'll definitely move there.

**Comment by RaduBerinde** [comment]
TFTR!

**Pull request merged**